### DISCUSSION TITLE
I think my UI library (which uses the same organization as the design system template) is not tree shakeable by Next.js

### DISCUSSION THREAD

**Discussion by williamlmao** [vercel/turborepo#3665] [Help] [answered]
[Working example/Source](https://github.com/williamlmao/chimera-tw)

I noticed that every time I added new components to my ui library, the bundle sizes of the pages all grew larger. 

So I tested named vs default exports in a few pages, and here's what I found. 

docs:build: ├ ○ /docs/MuiTreeShakeTest1                3.38 kB         198 kB  (imported a button + slider from MUI, using default imports)
docs:build: ├ ○ /docs/MuiTreeShakeTest2 (673 ms)       5.57 kB         200 kB (imported a button + slider + switch from MUI, using named imports)
docs:build: ├ ○ /docs/MuiTreeShakeTest3 (655 ms)       3.36 kB         190 kB.  (imported a button  from MUI, using named imports)
docs:build: ├ ○ /docs/TestPopOnly                      1.09 kB         213 kB  (Imported a only Popover from my UI library using named import)
docs:build: ├ ○ /docs/TestPopWithOther                 1.57 kB         213 kB (Imported a only Popover + 3 other components from my UI library using named import)

You can see that in TestPopOnly and TestPopWithOther, the bundle size was the same despite different amount of components being imported. 

It looks like tree shaking was working fine with MUI, so I don't think it's a problem with nextjs. 

I have kept up the setup from the [turborepo design system](https://github.com/vercel/turbo/tree/main/examples/design-system) mostly the same. 

A ui package, with all components being exported from index.ts in /src. The only thing I really changed was updating tsup and adding the `--treeshake` flag to it, but that didn't do much. 

Any help would be appreciated!

**Comment by anthonyshew** [comment, marked as answer]
We've updated our examples to be treeshakable by default. Check them out: https://github.com/vercel/turbo/tree/main/examples
